cmd/cstor-pool-mgmt/controller/replica-controller/new_replica_controller.go

```diff
@@ -114,7 +114,14 @@ func NewCStorVolumeReplicaController(
 			q.Operation = common.QOpAdd
 			glog.Infof("cStorVolumeReplica Added event : %v, %v", cVR.ObjectMeta.Name, string(cVR.ObjectMeta.UID))
 			controller.recorder.Event(cVR, corev1.EventTypeNormal, string(common.SuccessSynced), string(common.MessageCreateSynced))
-			cVR.Status.Phase = apis.CVRStatusPending
+
+			// For New request phase of cVR will be empty
+			if IsEmptyStatus(cVR) {
+				cVR.Status.Phase = apis.CVRStatusPending
```

Comment:
instead of modifying 'Status' which impacts the state diagram, how about using some annotations on CVR? This can probably help for our upgrade as well. Is this possible? cc: @AmitKumarDas